render/helpers.go

```diff
@@ -172,12 +183,15 @@ func Slug(s string) string {
 	return slug.Make(s)
 }
 
-// LegacyDataSetDownloadURI builds a URI string for a legacy dataset download URI.
-func LegacyDataSetDownloadURI(pageURI, filename string) string {
+func LegacyDataSetDownloadURI(input ...string) string {
```

Comment:
So instead of `<a href="{{legacyDatasetDownloadURI .URI}}"` have `<a href="{{print .URI}}"`?

Reply:
Print is used to concatenate strings together in the template. If `.URI` is all you need then `<a href="{{.URI}}"` would suffice as `<a href="{{print .URI}}"` would do nothing extra.

In the example https://github.com/ONSdigital/dp-frontend-renderer/blob/d5d2e1e8ddbcc0007f10d60fc881877905c6472c/assets/templates/search/filter.tmpl#L22

We are taking the text `checkout-` and then appending the contents of what is stored in the variable `$content.Type` then storing that output into a new variable called `id` which can be used later in the template, e.g. `<input id=$id` or 
`<label for=$id>`

It looks like you are expecting the contents of `.URI` to be just the `%s` part of the string for
`URIFormatOneParameter` which is `/file?uri=%s`
So by removing the helper legacyDatasetDownloadURI and just using .URI you will not have the `/file?uri=` part.

So to concatenate the two from within the template using the generic helper that already exists (`print` rather than this new helper `legacyDatasetDownloadURI`) you would need to do something like `{{print "/file?uri=" .URI}}`
I believe you may need to store that in a variable before using e.g. `{{$yourVar := (print "/file?uri=" .URI)}}`
Then to use just do `<a href="{{.yourVar}}"` However see if you can do it without a variable first e.g. something like this: <a href="{{print "/file?uri=" .URI}}" (note may need to add parentheses)